src/MagnumPlugins/GltfImporter/GltfImporter.cpp

```diff
@@ -3286,15 +3286,40 @@ Containers::Optional<MeshData> GltfImporter::doMesh(const UnsignedInt id, Unsign
         return {};
     }
 
+    /* Morph target attributes */
+    const Utility::JsonToken* gltfTargets = gltfPrimitive.find("targets"_s);
+    if(gltfTargets) {
+        if(!_d->gltf->parseArray(*gltfTargets)) {
+            Error{} << "Trade::GltfImporter::mesh(): invalid targets property in mesh" << _d->gltfMeshPrimitiveMap[id].first();
+            return {};
+        }
+
+        for(Utility::JsonArrayItem gltfTarget: gltfTargets->asArray()) {
+            if(!_d->gltf->parseObject(gltfTarget)) {
+                Error{} << "Trade::GltfImporter::mesh(): invalid morph target" << gltfTarget.index() << "in mesh" << _d->gltfMeshPrimitiveMap[id].first();
+                return {};
+            }
+
+            for(Utility::JsonObjectItem gltfMorphAttribute: gltfTarget.value().asObject()) {
+                if(!_d->gltf->parseUnsignedInt(gltfMorphAttribute.value())) {
+                    Error{} << "Trade::GltfImporter::mesh(): invalid morph target attribute" << gltfMorphAttribute.key() << "in mesh" << _d->gltfMeshPrimitiveMap[id].first();
+                    return {};
+                }
+
+                arrayAppend(attributeOrder, InPlaceInit, gltfMorphAttribute.key(), gltfMorphAttribute.value().asUnsignedInt(), gltfTarget.index());
+            }
+        }
+    }
+
     /* Sort and remove duplicates except the last one. Attributes sorted by
-       name so that we add attribute sets in the correct order and can warn if
-       indices are not contiguous. */
+       name (per morph target) so that we add attribute sets in the correct
+       order and can warn if indices are not contiguous. */
     const std::size_t uniqueAttributeCount = stableSortRemoveDuplicatesToPrefix(arrayView(attributeOrder),
-        [](const Containers::Pair<Containers::StringView, UnsignedInt>& a, const Containers::Pair<Containers::StringView, UnsignedInt>& b) {
-            return a.first() < b.first();
+        [](const Containers::Triple<Containers::StringView, UnsignedInt, Int>& a, const Containers::Triple<Containers::StringView, UnsignedInt, Int>& b) {
+            return a.third() != b.third() ? a.third() < b.third() : a.first() < b.first();
         },
-        [](const Containers::Pair<Containers::StringView, UnsignedInt>& a, const Containers::Pair<Containers::StringView, UnsignedInt>& b) {
-            return a.first() == b.first();
+        [](const Containers::Triple<Containers::StringView, UnsignedInt, Int>& a, const Containers::Triple<Containers::StringView, UnsignedInt, Int>& b) {
+            return a.third() == b.third() && a.first() == b.first();
         });
```

Comment:
One more thing :sweat_smile: 

> The stable sort is adjusted to first sort on morph target id followed by attribute name, ensuring only duplicates *within* a morph target/base attributes are filtered and not *across* morph targets.

Good that you said that -- can you add an explicit test for this behavior? Should be enough to expand the existing `meshDuplicateAttributes()` case.